dll/win32/msafd/misc/sndrcv.c

```diff
@@ -514,59 +597,73 @@ WSPRecvFrom(SOCKET Handle,
                                     APCContext,
                                     IOSB,
                                     IOCTL_AFD_RECV_DATAGRAM,
-                                    &RecvInfo,
-                                    sizeof(RecvInfo),
+                                    RecvInfo,
+                                    sizeof(AFD_RECV_INFO_UDP),
                                     NULL,
                                     0);
-
-    /* Wait for completion of not overlapped */
-    if (Status == STATUS_PENDING && lpOverlapped == NULL)
+    if (Status == STATUS_PENDING)
     {
-        WaitForSingleObject(SockEvent, INFINITE); // BUGBUG, shouldn wait infinitely for receive...
+        /* Wait for completion of not overlapped */
+        if (!lpOverlapped)
+        {
+            /* BUGBUG, shouldn't wait infinitely for receive... */
```

Comment:
```suggestion
            /* FIXME: shouldn't wait infinitely for receive... */
```
FIXME is much more common terminology in ReactOS codebase than bugbug. Easier to grep for.